packages/core/src/Ponder.ts

```diff
@@ -72,6 +72,12 @@ export class Ponder {
     this.common = { options, logger, errors, metrics, telemetry };
 
     this.buildService = new BuildService({ common: this.common });
+
+    // Check Nodejs version
+    if (Number(process.version.split(".")[0].slice(1)) < 18)
+      this.common.logger.fatal({
+        msg: `Node version:${process.version} does not meet the >=18 requirement`,
+      });
```

Comment:
Two things:
1. Will this work on Bun?
2. I'm not sure that the process would ever make it to this line in practice, because the error could (probably does!) happen while executing dependency code. So, it might make sense to do this in `bin/ponder.ts` before doing basically anything else.